sdv/single_table/base.py

```diff
@@ -42,3 +46,175 @@ def get_parameters(self):
     def get_metadata(self):
         """Return the ``SingleTableMetadata`` for this synthesizer."""
         return self.metadata
+
+    def _validate_metadata_matches_data(self, columns):
+        errors = []
+        metadata_columns = self.metadata._columns if self.metadata._columns else []
+        invalid_data_columns = set(columns).difference(metadata_columns)
+        if invalid_data_columns:
+            errors.append(
+                f'The columns {sorted(invalid_data_columns)} are not present in the metadata.')
+
+        invalid_metadata_columns = set(metadata_columns).difference(columns)
+        if invalid_metadata_columns:
+            errors.append(
+                f'The metadata columns {sorted(invalid_metadata_columns)} '
+                'are not present in the data.'
+            )
+
+        if errors:
+            raise InvalidDataError(errors)
+
+    def _get_primary_plus_alternate_keys(self):
+        keys = set()
+        if self.metadata._primary_key:
+            keys = {self.metadata._primary_key}
+
+        if self.metadata._alternate_keys:
+            keys.update(set(self.metadata._alternate_keys))
+
+        return keys
+
+    def _get_sequence_keys(self):
```

Comment:
can we add to the name that it returns a set? Otherwise I would expect it to just return whatever `self._sequence_key` is